silkworm/rpc/core/estimate_gas_oracle.cpp

```diff
@@ -69,6 +69,12 @@ Task<intx::uint256> EstimateGasOracle::estimate_gas(const Call& call, const silk
                       << ", allowance " << allowance;
             hi = uint64_t(allowance);
         }
+    } else {
+        const auto base_fee_per_gas = block.header.base_fee_per_gas;
+        if (base_fee_per_gas) {
+            std::cout << "set gas_price to: " << intx::to_string(*base_fee_per_gas);
+            call.set_gas_price(*base_fee_per_gas);
+        }
```

Comment:
Here I would just overwrite `gas_price` value with `block.header.base_fee_per_gas` (you can use `value_or` for this) and then always pass `gas_price` as overridden gas price into `call.to_transaction`.

Reply:
ok